web/src/lib/taxation/helpers.ts

```diff
@@ -0,0 +1,58 @@
+import { postTaxFilingsOnboarding } from "@/lib/api-client/apiClient";
+import { UpdateQueue } from "@/lib/types/types";
+import { getCurrentBusiness } from "@businessnjgovnavigator/shared/index";
+import { LookupLegalStructureById } from "@businessnjgovnavigator/shared/legalStructure";
+import { ProfileData } from "@businessnjgovnavigator/shared/profileData";
+const displayBusinessName = (profileData: ProfileData): boolean => {
+  return LookupLegalStructureById(profileData.legalStructureId).elementsToDisplay.has("businessName");
+};
+const displayResponsibleOwnerName = (profileData: ProfileData): boolean => {
+  return LookupLegalStructureById(profileData.legalStructureId).elementsToDisplay.has("responsibleOwnerName");
+};
+interface Props {
+  updateQueue: UpdateQueue;
+  optionalProfileDataIfSeparateContext?: ProfileData;
+}
+// the aim of this fn is to hit postTaxFilingsOnboarding and staged update queue
+export const toBeNamed = async (props: Props) => {
+  const currentBusiness = props.updateQueue.currentBusiness();
+  const updateQueue = props.updateQueue;
+  const profileData = props.optionalProfileDataIfSeparateContext || currentBusiness.profileData;
+  let encryptedTaxIdToSubmitToTaxApi = "";
+  let businessNameToSubmitToTaxApi = "";
+  const taxIdToSubmitToTaxApi = profileData.taxId || "";
+  const displayBusinessNameValue = displayBusinessName(profileData);
+  const displayResponsibleOwnerNameValue = displayResponsibleOwnerName(profileData);
+  if (profileData.taxId === currentBusiness.profileData.taxId) {
+    encryptedTaxIdToSubmitToTaxApi = profileData.encryptedTaxId || "";
+  }
+  if (displayBusinessNameValue) {
+    businessNameToSubmitToTaxApi = profileData.businessName;
+  }
+  if (displayResponsibleOwnerNameValue) {
+    businessNameToSubmitToTaxApi = profileData.responsibleOwnerName;
+  }
+  if (!businessNameToSubmitToTaxApi) return;
```

Comment:
[sand] This line is essentially a guard clause, but it's at the end of a huge block of code. The way I'm reading this is there's a whole bunch of setup and then a condition that prevents further things from happening. I'd move this up to the top, above all your logical operators, and maybe even above some of your variable declarations to be more prominent.

Moving this to the top takes some of the mental toll off of the person who's reading this later, because you're making it explicitly clear "if this... then nothing" instead of requiring them to keep a ton of information in their head and then showing them this condition.

Reply:
In order to make this more readable, I have moved a similar check to right below checking for the businessName or responsibleOwnerName in:

```
  const displayBusinessNameValue = displayBusinessName(profileData);
  const displayResponsibleOwnerNameValue = displayResponsibleOwnerName(profileData);

  if (!displayBusinessNameValue && !displayResponsibleOwnerNameValue) return;
  ```

That way if both values are blank/empty strings it will return.